python/python/bystro/search/utils/annotation.py

```diff
@@ -1,44 +1,89 @@
+    args: str
+    program: str
+
+
+class StatisticsOutputExtensions(Struct, frozen=True):
+    json: str = "statistics.json"
+    tsv: str = "statistics.tsv"
+    qc: str = "statistics.qc.tsv"
+
+
+class StatisticsConfig(Struct, frozen=True):
+    dbSNPnameField: str = "dbSNP.name"
+    siteTypeField: str = "refSeq.siteType"
+    exonicAlleleFunctionField: str = "refSeq.exonicAlleleFunction"
+    refField: str = "ref"
+    homozygotesField: str = "homozygotes"
+    heterozygotesField: str = "heterozygotes"
+    altField: str = "alt"
+    programPath: str = "bystro-stats"
+    outputExtensions: StatisticsOutputExtensions = StatisticsOutputExtensions()
+
+    @staticmethod
+    def from_dict(annotation_config: dict[str, Any] | None = None):
+        """Get statistics config from a dictionary"""
+        if annotation_config is None:
+            return StatisticsConfig()
+
+        stats_config: Optional[dict[str, Any]] = annotation_config.get("statistics")
+
+        if stats_config is None:
+            return StatisticsConfig()
+
+        if "outputExtensions" in stats_config:
+            stats_config["outputExtensions"] = StatisticsOutputExtensions(
+                **stats_config["outputExtensions"]
+            )
+
+        return StatisticsConfig(**stats_config)
+
+
 class StatisticsOutputs(Struct, frozen=True):
     """
-        Paths to all possible Bystro statistics outputs
-
-        Attributes:
-            json: str
-                Basename of the JSON statistics file
-            tab: str
-                Basename of the TSV statistics file
-            qc: str
-                Basename of the QC statistics file
+    Paths to all possible Bystro statistics outputs
+
+    Attributes:
+        json: str
+            Basename of the JSON statistics file
+        tab: str
+            Basename of the TSV statistics file
+        qc: str
+            Basename of the QC statistics file
     """
+
     json: str
     tab: str
     qc: str
 
 
 class AnnotationOutputs(Struct, frozen=True):
     """
-        Paths to all possible Bystro annotation outputs
-        
-        Attributes:
-            output_dir: str
-                Output directory
-            archived: str
-                Basename of the archive
-            annotation: str
-                Basename of the annotation TSV file, inside the archive
-            sampleList: Optional[str]
-                Basename of the sample list file, inside the archive
-            log: Basename of the log file, inside the archive
-            statistics: Optional[StatisticsOutputs]
-                Basenames of the statistics files, inside the archive
+    Paths to all possible Bystro annotation outputs
+
+    Attributes:
+        output_dir: str
+            Output directory
+        archived: str
+            Basename of the archive
+        annotation: str
+            Basename of the annotation TSV file, inside the archive
+        sampleList: Optional[str]
```

Comment:
Agreed. The elegance of this system is that our YAML configuration maps 1:1 with source code. I would like to avoid the bookkeeping of needing to rewrite these values on a case by case basis, so instead made a ticket to update the YAML configuration to use snake_case directly: https://github.com/bystrogenomics/bystro/issues/328

Reply:
it's what the youth crave, true